docs/dev/new_check_howto.md

```diff
@@ -0,0 +1,353 @@
+    pytest -v -m"integration"
+
+[testenv:flake8]
+skip_install = true
+deps = flake8
+commands = flake8 .
+
+[flake8]
+exclude = .eggs,.tox
+max-line-length = 120
+```
+
+You might notice that when invoking `pytest` we now pass an extra argument,
+`-m"integration"` in one case and `-m"not integration"` in another. These are
+called _attributes_ in pytest terms and `-m"integration"` will tell pytest to
+only run tests that are marked with the `integration` attribute, or the other
+way around if `not integration` is specified.
+
+Let's add the integration test to our `my_check/tests/test_check.py` module:
+
+```python
+import subprocess
+import os
+import time
+from datadog_checks.utils.common import get_docker_hostname
+
+
+@pytest.mark.integration
+def test_service_check(aggregator):
+    c = MyCheck('my_check', {}, {}, None)
+
+    HERE = os.path.dirname(os.path.abspath(__file__))
+    args = [
+        "docker-compose",
+        "-f", os.path.join(HERE, 'docker-compose.yml')
+    ]
+
+    # start the Nginx container
+    subprocess.check_call(args + ["up", "-d"])
+    time.sleep(5)  # we should implement a better wait strategy :)
+
+    # the check should send OK
+    instance = {
+        'url': 'http://{}:8000'.format(get_docker_hostname()),
+        'search_string': "Thank you for using nginx."
+    }
+    c.check(instance)
+    aggregator.assert_service_check('my_check.all_good', MyCheck.OK)
+
+    # the check should send WARNING
+    instance['search_string'] = "Apache"
+    c.check(instance)
+    aggregator.assert_service_check('my_check.all_good', MyCheck.WARNING)
+
+    # stop the container
+    subprocess.check_call(args + ["down"])
+
+    # the check should send CRITICAL
+    c.check(instance)
+    aggregator.assert_service_check('my_check.all_good', MyCheck.CRITICAL)
+```
+
+Again, we could run all the tests but let's run only the integration tests for
+faster iterations:
+
+    tox -e integrations
+
+Our check is almost done, let's add the final touches.
+
+## Final touches
+
+### Fill the README
+
+The readme file as provided by our cookiecutter template has already the right
+format, all you need to do is filling out the relevant sections, in particular
+you're supposed to write something in place of the `[...]` strings you'll find
+in the file.
+
+### Add images and logos
+
+This is the expected directory structure for images and logos:
+
+    my_check/
+    ├── images
+    │   └── an_awesome_image.png
+    └── logos
+        ├── avatars-bot.png
+        ├── saas_logos-bot.png
+        └── saas_logos-small.png
+
+The `images` folder is meant to contain any images that are needed in the
+integration tile. To be used as such, they should be referenced in the `## Overview`
+and/or `## Setup` sections in `README.md` as markdown images using their public URLs.
+Because the integrations-core and integrations-extras repositories are public,
+a public URL can be gotten for any of these files via `https://raw.githubusercontent.com`.
```

Comment:
`obtained` sounds better IMO

Reply:
OMG it's ugly, in my defense that comes from another source